### DISCUSSION TITLE
lint and fix rust code

### DISCUSSION THREAD

**Discussion by juli1** [zed-industries/zed#11177] [Feature Requests]
When using `zed` with Rust, I want to be able to

1. Auto-format on save
2. Fix any issue using `rustfmt` or `clippy`

I saw how to auto-format (assing `"format_on_save": "on",`) but nothing on fixing the issue.

An additional ask would be to be able to configure the settings using a UI (like VSCode does). This would be very helpful.

**Comment by onx2** [comment]
Do these docs help answer? https://zed.dev/docs/configuring-zed#lsp

Or maybe something like this would work?
```
"format_on_save": {
  "external": {
    "command": "cargo clippy --fix",
    "arguments": [... any args you need]
  }
}
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
